ui/apps/platform/src/Containers/Violations/Details/DeploymentDetails.tsx

```diff
@@ -176,12 +186,27 @@ const DeploymentDetails = ({ alertDeployment }: DeploymentDetailsProps) => {
                                             </Tr>
                                         </Thead>
                                         <Tbody>
+                                            {selectedNetworkPolicy && (
```

Comment:
Can you show this scenario for conditional and to render modal at a PatternFly Developer Office Hours?

When I showed an example recently, the team said they had not been aware of many uses cases.

Reply:
I had brought this up on the PF Slack a while back and they said that either method would be fine. In the implementation for `Modal`, if `isOpen` is set to false the component renders `null` so either scenario would end up with the same component tree when the Modal is closed.

The reason for the conditional rendering here is so that `selectedNetworkPolicy` can be inferred as non-null when inside the modal, avoiding the need to muddy the code of that component with optional chaining checks, default values, etc., when logically there is no way for the passed policy to be null.

Reply:
Yes, that is the root of the issue that I had. TypeScript understands conditional and, but not conditional rendering via `isOpen` prop. When I asked whether it was a pattern to document, there was some hesitation.